### DISCUSSION TITLE
vite refreshing page if changes are made to server code.

### DISCUSSION THREAD

**Discussion by salman3001** [adonisjs/core#4526] [Help] [answered]
Hi, I have migrated a nuxt project to invertia ssr with vue 3. everythig is working good so far. the problem is that project have so many assets.  so full refresh on page takes some time. 

whenever i make changes to server code. vite hard reload the page. i dont know if it was supposed to be like this. Please if any one knows how to solve this issue.

**Comment by thetutlage** [comment]
I think Vite should not hard reload the page when you make changes in your frontend code. Can you please share a reproduction repo for the same that we can use to debug the issue?

**Reply by salman3001** [reply]
hi, @thetutlage , thanks for reply. 

here is reproduction repo  --> https://github.com/salman3001/adonis-vite-refresh-issue-reproduction.git

this is fresh install of inertia starter kit. run the dev server and visit  "/inertia". now modify the App->model->user.ts file.  you will see the full reload of page. it is quick because there is only one page. in my case there many components, ui library, sass files, icon  libraries which takes some time to reload again. 

 now make changes to inertia->pages->home.vue. now vite will not full reload but page will update.  that means if vue files are modified there is no full reload. but if file from server code is changed like controllers, models etc then there is full reload.   please if this can be solved. becuase it is monorepo so for sure we will be working on server and frontend together.

**Reply by Julien-R44** [reply]
@salman3001 See https://github.com/adonisjs/core/discussions/4526#discussioncomment-9139957

It should solve your issue

**Comment by thetutlage** [comment]
Just read the title carefully. It says backend code :). Yes, that is expected as of now.

**Reply by Julien-R44** [reply, marked as answer]
Also, you can try the new experimental HMR support we released some days ago. Follow this : https://github.com/adonisjs/core/releases/tag/v6.7.0

It should solve the issue you are facing

**Reply by salman3001** [reply]
hi, @Julien-R44  ,  i tried it. thanks. now vite is not reloading if backend code is modified. but on some files it is still reloading.  like i tried to modify the middleware file which caused full reload. can we control this? 

i tried updating the below options

`  "hotHook": {
    "boundaries": [
      "../app/controllers/**/*.ts"
    ]
  }`

but still there is reload.

**Reply by salman3001** [reply]
ok... it has solved  many issues now. thanks
